mvrx/src/main/kotlin/com/airbnb/mvrx/BaseMvRxViewModel.kt

```diff
@@ -223,6 +263,29 @@ abstract class BaseMvRxViewModel<S : MvRxState>(
             .disposeOnClear()
     }
 
+    /**
+     * Similar to [execute]. This subscribes to an Observable, provides a reducing function to update state when the observable passes a value,
+     * and registers the subscription to be disposed when the ViewModel is cleared.
+     *
+     * This function should be preferred over calling [Observable.subscribe] directly so that disposal is handled for you, and so setting state
+     * is simplified.
+     */
+    fun <T> Observable<T>.executeWithoutAsync(stateReducer: S.(value: T) -> S): Disposable {
+        val blockExecutions =
+            config.currentMockBehavior?.blockExecutions ?: MockBehavior.BlockExecutions.No
+
+        if (blockExecutions != MockBehavior.BlockExecutions.No) {
```

Comment:
Is there anyway we can handle this in the state store? It seems like the concept of blocking etc. is a state store concept, not a view model concept.